Add ServiceCategory spec (#118)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: supporting_content/dimensions/servicecategory.md

# Column: ServiceCategory

## Example provider mappings

Current column mappings found in available data sets:

| Provider  | Data set                 | Column                                     |
| --------- | ------------------------ | ------------------------------------------ |
| AWS       | CUR                      | None                                       |
| GCP       | Big Query Billing Export | None                                       |

> | Microsoft | Cost details             | ServiceFamily is close, but based on usage |

Reviewer: not sure if, in case of Azure, serviceName is a good match for providers mapping, especially if use consumedService for serviceName mapping, which I like

Author: ServiceFamily is technically the right answer. The difference is just that the values will need to change in some cases based on the resource type.

## Discussion / Scratch space

Principles behind Service Category, granularity, when to add, when to split etc.

- Every category must represent a single technology. Do not "and" multiple categories together unless they are synonymous.
- Each category should be unique and should not be a logical child of another category (e.g., Containers are a subclassification of Compute).
- Categories are used for cost classification purposes only. Do not create new categories for marketing differentiation (e.g., Containers).
- Avoid unnecessary words. Prefer short, concise names.

> - Each service should have 1 and only 1 category that best aligns to its architectural design.

Author: ```suggestion
- Each service should have 1 and only 1 category that best aligns to its purpose.
```

> - Each service should have 1 and only 1 service category that best aligns to its architectural design.

Reviewer 2: When referring to this column shouldn't it use the Proper Name? e.g.  Service Category instead of service category

> - Each service should have 1 and only 1 service category that best aligns to its architectural design.

Reviewer 2: ```suggestion
- Each service should have 1 and only 1 Service Category that best aligns to its architectural design.
```

- Each category should have around 3-10 services for the major cloud providers. Avoid categories with only 1 or with over 20 services. No category should have more than 20% of services.
- Do not create categories that describe traits of the service that could be applied to many categories (e.g., "serverless").
- Beyond purely cost categorization purposes, the category is based on the primary function the service intends to solve.

> Note: While a service can only be in one category, usage/cost for that service may include charges related to another category. For example, resources under the Databases service category can have storage and compute cost - however, the primary function of the overall service is to provide database functionality therefore it’s considered to be under the database category.

Reviewer 2: When referring to this column shouldn't it use the Proper Name? e.g. Service Category instead of service category
```suggestion
Note: While a service can only be in one category, usage/cost for that service may include charges related to another category. For example, resources under the Databases Service Category can have storage and compute cost - however, the primary function of the overall service is to provide database functionality therefore it’s considered to be under the database category.
```

Cloud provider mappings:

> | Category                                              | AWS (link)                               | GCP (link)                    | Microsoft (link)                  | Alibaba Cloud (link)                 | Oracle Cloud (OCI)             |

Reviewer 2: Suggested link for Microsoft:  https://learn.microsoft.com/en-us/azure/cost-management-billing/automate/understand-usage-details-fields  (Cannot find a more detailed page just for Service Family)
```suggestion
| Category                                              | AWS (link)                               | GCP (link)                    | Microsoft ([link](https://learn.microsoft.com/en-us/azure/cost-management-billing/automate/understand-usage-details-fields))                  | Alibaba Cloud (link)                 | Oracle Cloud (OCI)             |
```
